Source/msg.cpp

```diff
@@ -2397,12 +2397,12 @@ void DeltaLoadLevel()
 	uint8_t localLevel = GetLevelForMultiplayer(*MyPlayer);
 	DLevel &deltaLevel = GetDeltaLevel(localLevel);
 	if (leveltype != DTYPE_TOWN) {
-		for (int i = 0; i < ActiveMonsterCount; i++) {
+		for (int i = 0; i < MaxMonsters; i++) {
 			if (deltaLevel.monster[i].position.x == 0xFF)
 				continue;
 

```

Comment:
Isn't this also needed?
```suggestion
		ActiveMonsterCount = 0;
		for (int i = 0; i < MaxMonsters; i++) {
			if (deltaLevel.monster[i].position.x == 0xFF)
				continue;

			ActiveMonsterCount++;
```

Reply:
in conclusion adding `ActiveMonsters[]` would have made the function more correct/safe and only iterate over the spawned monsters.

Using `MaxMonsters` also make the function correct, and all unspawned monsters will simply be skipped as they are `0xFF` in the delta array.